frontend/src/features/public-form/components/FormEndPage/components/EndPageBlock.tsx

```diff
@@ -78,7 +78,11 @@ export const EndPageBlock = ({
       <Box mt="2.25rem">
         <Button
           as="a"
-          href={endPage.buttonLink ?? window.location.href}
+          href={
+            endPage.buttonLink?.length
+              ? endPage.buttonLink
+              : window.location.href
+          }
```

Comment:
The null coalescing operator was too strict, but this case is well suited for the [logical or](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR) instead. (`a || b`, returns `b` is `a` is falsy, including being empty string)

```suggestion
          href={endPage.buttonLink || window.location.href}
```